core/src/main/java/io/temporal/samples/hello/HelloSignalWithTimer.java

```diff
@@ -0,0 +1,167 @@
+/*
+ *  Copyright (c) 2020 Temporal Technologies, Inc. All Rights Reserved
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.hello;
+
+import io.temporal.client.WorkflowClient;
+import io.temporal.client.WorkflowOptions;
+import io.temporal.client.WorkflowStub;
+import io.temporal.failure.CanceledFailure;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import io.temporal.worker.Worker;
+import io.temporal.worker.WorkerFactory;
+import io.temporal.workflow.*;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+
+/**
+ * Sample Temporal workflow that shows receiving signals for a specific time period and then process
+ * last one received and continueasnew.
+ */
+public class HelloSignalWithTimer {
+  static final String TASK_QUEUE = "HelloSignalWithTimerTaskQueue";
+  static final String WORKFLOW_ID = "HelloSignalWithTimerWorkflow";
+
+  @WorkflowInterface
+  public interface SignalWithTimerWorkflow {
+    @WorkflowMethod
+    void execute();
+
+    @SignalMethod
+    void doUpdate(String value);
```

Comment:
nit: Can we not call this update since `Temporal` now has a concept called `update` that I wouldn't want to confuse with this.